lib/ecto_ldap/adapter.ex

```diff
@@ -122,26 +176,26 @@ defmodule Ecto.Ldap.Adapter do
 
   def construct_scope(_), do: {:scope, :eldap.wholeSubtree}
 
-  def construct_attributes(_) do
-  end
-
-  #   def extract_parameter(wheres) do
-  #     [{{_, _, [_ | parameter]}, _, _} | _] = extract_array(wheres)
-  #     case parameter do
-  #       {:&, [], [0]} -> []
-  #       _ -> to_string(hd(parameter))
-  #     end
-  #   end
-
-  def construct_attributes(%Ecto.Query.SelectExpr{fields: fields}) do
-    case fields do
-      [{:&, [], [0]}] -> []
-      _ -> fields
+  def construct_attributes(%{select: select}) do
+    case select.fields do
+      [{:&, [], [0]}] -> 
+        nil
+      attributes -> 
+        {
+          :attributes,
+          attributes
+            |> Enum.map(&extract_select/1)
+            |> Enum.map(&convert_to_erlang(&1))
+        }
     end
   end
 
-  def prepare(:update_all, query), do: raise UndefinedFunctionError, "Update is currently unsupported"
-  def prepare(:delete_all, query), do: raise UndefinedFunctionError, "Delete is currently unsupported"
+  def extract_select({{:., _, [{:&, _, _}, select]}, _, _}), do: select
+
+  def convert_to_erlang(list) when is_list(list), do: Enum.map(list, &convert_to_erlang/1)
+  def convert_to_erlang(string) when is_binary(string), do: :binary.bin_to_list(string)
+  def convert_to_erlang(atom) when is_atom(atom), do: atom |> Atom.to_string |> convert_to_erlang
+  def convert_to_erlang(num) when is_number(num), do: num
```

Comment:
aww yiss -- erlang conversions

the `convert_from_erlang` methods are down around L230:237